### PR TITLE
Basic Canton HA content edited for readability

### PR DESCRIPTION
**Pull request by da-katmurp**

### CODE REVIEW

**Comment by da-katmurp** [comment]
Hi @soren-da I have tagged you as a reviewer on this PR.  I'm aware of some of this content being out of date so I'm looking for a currency and correctness review. Thank you.

File: docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst [hunk 1]
```diff
@@ -1,142 +1,94 @@
...
 
 .. _participant-ha-system-topology:
 .. https://lucid.app/lucidchart/cd96a3a6-e10b-4edc-bfb3-a70e484d7c06
 .. image:: participant-ha-system-topology.svg
    :align: center
    :width: 100%
 
-Why a Shared Database?
-""""""""""""""""""""""
+The load balancer configuration contains details of all Ledger API server addresses and the ports for the participant node replicas. Replicas expose their active or passive status via a health endpoint.
```
Comment (soren-da) [inline docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst:28]: this section is slightly out of date. now the ledger API server is shutdown when a node goes from active to passive, so the load balancer will also detect that the backend port becomes unreachable.

File: docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst [hunk 2]
```diff
@@ -1,142 +1,94 @@
...
-""""""""""""""""""""""""""""""""""""""""""""""""""
-
-A participant node replica tries to acquire an exclusive application level lock
-(e.g. `Postgres advisory lock
-<https://www.postgresql.org/docs/11/explicit-locking.html#ADVISORY-LOCKS>`_)
-bound to a particular database connection and use that same connection for all
-writes that are not idempotent. The replica that has acquired the lock is the
-leader and the active replica. Using the same connection for writes ensures that
-the lock is held while writes are performed.
+A participant node replica acquires an exclusive application-level lock (e.g. `Postgres advisory lock <https://www.postgresql.org/docs/11/explicit-locking.html#ADVISORY-LOCKS>`_) which is bound to a particular database connection. The active replica that acquires the lock becomes the leader. The replica then uses the same connection for all writes that are not idempotent.
```
Comment (soren-da) [inline docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst:58]: slightly outdated: The replica uses a write connection pool that is tied to an exclusive lock on a main connection, and a shared lock on all pool connections. If the main lock is lost, the pool's connections are ramped down. the new active replica needs to wait until all the passive node's pool connections are closed, which is done by trying to acquire the shared lock in exclusive mode

File: docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst [hunk 3]
```diff
@@ -1,142 +1,94 @@
...
+To avoid passive replicas attempting to write to the database - any such attempt fails and produces an error - we use a coarse-grained guard on domain connectivity and API services.
+
+To prevent the passive replica from processing domain events, and ensure it rejects incoming Ledger API requests, we keep the passive replica disconnected from the domains as a coarse-grained enforcement.
+
+Lock Loss and Failover
+""""""""""""""""""""""
+
+If the active replica crashes or loses connection to the database, the lock is released and a passive replica can claim the lock and become active. Any pending writes in the formerly active replica fail due to losing the underlying connection and the corresponding lock.
+
+The active replica has a grace period in which it may rebuild the connection and reclaim the lock to avoid unnecessary failover on short connection interruptions.
```
Comment (soren-da) [inline docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst:90]: slightly outdated: the grace period now comes from the higher frequency of checking the health of the lock in the active replica vs. trying to acquire the lock in the passive replica at a lower frequency.

File: docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst [hunk 4]
```diff
@@ -1,142 +1,94 @@
...
+Lock Loss and Failover
+""""""""""""""""""""""
+
+If the active replica crashes or loses connection to the database, the lock is released and a passive replica can claim the lock and become active. Any pending writes in the formerly active replica fail due to losing the underlying connection and the corresponding lock.
+
+The active replica has a grace period in which it may rebuild the connection and reclaim the lock to avoid unnecessary failover on short connection interruptions. 
+
+The passive replicas continuously attempt to acquire the lock within a configurable interval. Once the lock is acquired, the participant replication manager sets the state of the successful replica to active.
+
+When a passive replica becomes active, it connects to previously connected domains to resume event processing. The new active replica accepts incoming requests, e.g. on the Ledger API. The former active replica, that is now passive, rejects incoming requests as it can no longer write to the shared database.
```
Comment (soren-da) [inline docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst:94]: the ledger API server is started when a node becomes active. the formerly active replica that became passive will shut down its ledger API server to stop accepting requests

**Review by soren-da — APPROVED** [review]

**Pull request merged**